src/popup/components/Protocol/EnsureUserSavedMnemonic.tsx

```diff
@@ -47,28 +55,56 @@ const WordButton = (props: {
   onClick: () => void;
 }) => {
   return (
-    <Button disabled={!props.isEnabled} onClick={props.onClick}>
+    <Button
+      className={(() => {
+        return !props.isEnabled ? "ClickedWord" : "";
+      })()}
+      onClick={props.onClick}
+    >
       {props.word}
     </Button>
   );
 };
 
-export function EnsureUserSavedMnemonic(props: { onComplete: () => void }) {
-  const [buttons, setButtons] = React.useState<CheckButton[]>([]);
-  const [mnemonicArr, setMnemonicArr] = React.useState<string[]>([]);
+const TextField = (props: { input: string }) => {
+  return <TextArea defaultValue={props.input} />;
+};
 
-  const [counter, setCounter] = React.useState(0);
+export function EnsureUserSavedMnemonic(props: { onComplete: () => void }) {
```

Comment:
I think the logic for this component is getting complex enough to warrant extracting a (unit tested) class to manage  this state. Could either be functional (as in returns a new instance each time a mutation happens) or could have a subscription for when mutations happen that triggers a re-render.